doc: add early warning for policy-default-local-cluster (#40108)
cilium-team/cilium: Documentation/network/clustermesh/policy.rst

Note that by default policies automatically select endpoints from all the clusters unless it is explicitly specified.
To restrict endpoint selection to the local cluster by default you can enable the option ``--policy-default-local-cluster``
via the ConfigMap option ``policy-default-local-cluster`` or the Helm value ``clustermesh.policyDefaultLocalCluster``. Changing

> this option will be a **breaking change** for existing policies. You can use the cilium cli to discover which policies are affected with

Reviewer: ```suggestion
this option is a **breaking change** for existing policies. You can use the cilium cli to discover which policies are affected with
```

> the command ``cilium clustermesh inspect-policy-default-local-cluster --all-namespaces`` (you can also replace ``--all-namespaces``
> by ``-n my-namespace`` if you want to only inspect policies from a particular namespace).

Reviewer: ```suggestion
the command ``cilium clustermesh inspect-policy-default-local-cluster --all-namespaces``. You can also replace ``--all-namespaces``
with ``-n my-namespace`` if you want to only inspect policies from a particular namespace.
```

Reviewer: Does this command provide the mitigations? If yes, we could add a small sentence at the end, something like:

> The output of this command is a set of recommendations in order to migrate your network policies to minimize disruption when changing the setting.

Reviewer: Structurally this feels a little bit like the wrong place to find these docs. A brand new user may click through the network policies, go into this section to configure policies between clusters, then there's this "legacy" upgrade instruction telling them to run some commands to migrate something. This is unrelated to the topic in this section of the docs.

At the end of the upgrade guide, [there is an advanced section](https://docs.cilium.io/en/latest/operations/upgrade/#advanced) with some subsections on "Migrating ..." different steps. Ideally we would have a consistent template for these sections, but we don't so far. Regardless, my suggestion would be to create a new section there that describes migrating the policy-default-local-cluster setting. Structurally it could have a brief description of the problem, behavior before, behavior after, then a subsubsection about "Mitigating the issue" which includes this example migration command.

Then from this policy doc, we could keep this aside about how the semantics of this statement are affected by the policy-default-local-cluster argument, then just include a simple link `Modifying this setting is a *breaking change*. To migrate this setting safely, read :ref:...`

Author: I pretty much did as you suggested I think, let me know if you have comments about the new section!
